source/Applications/Advanced/CreateDepthMap/CreateDepthMap.cpp

```diff
@@ -15,32 +15,26 @@ This example shows how to convert point cloud from ZDF file to OpenCV format, th
 
 namespace
 {
-    enum class Axis
-    {
-        x,
-        y,
-        z
-    };
-
-    template<Axis axis>
-    float getValue(const Zivid::PointXYZ &p);
-
-    template<>
-    float getValue<Axis::z>(const Zivid::PointXYZ &p)
+    float getValue(const Zivid::PointZ &p)
     {
         return p.z;
     }
 
-    template<Axis axis>
-    bool isLesserOrNan(const Zivid::PointXYZ &a, const Zivid::PointXYZ &b)
+    bool isLesserOrNan(const Zivid::PointZ &a, const Zivid::PointZ &b)
     {
-        return getValue<axis>(a) < getValue<axis>(b) ? true : std::isnan(getValue<axis>(a));
+        if(std::isnan(getValue(a)) && std::isnan(getValue(b)))
+        {
+            return false;
+        }
```

Comment:
What case do you cover by adding this?

Reply:
Some compilers dont like to compare a NaN value against another NaN value. This was the error Torbjørn and I got when we were running the sample in debug mode. This little if statement solves it